javascript/canvas.js

```diff
@@ -1,31 +1,151 @@
+      this.context.font ='30px monospace'
+      this.context.fillText(this.secretWord[index[i]], positionXLetter, initialPositionYLetter)
+    }
+
   }
 
+
   writeWrongLetter(letter, errorsLeft) {
-    // ... your code goes here
+    // console.log(' I have been called')
+    const initialPositionX = 700
+    const initialPositionY = 25
+
+    const lettersAsString = letter.join('') 
+
+    this.context.font ='30px monospace'
+    this.context.fillText(lettersAsString, initialPositionX, initialPositionY)
   }
 
   drawHangman(errorsLeft) {
-    // ... your code goes here
+    // console.log('yes, I was called')
+    const initialPositionX = 200
+    const initialPositionY = 400
+
+    const errors = errorsLeft
+
+    switch(errors) {
+      case 9:
+        this.context.beginPath()
+        this.context.moveTo(200, 400)
+        this.context.lineTo(300, 400)
+        this.context.stroke()
+        this.context.lineTo(250, 350)
+        this.context.stroke()
+        this.context.lineTo(200, 400)
+        this.context.stroke()
+        break;
+      case 8:
+        this.context.beginPath()
+        this.context.moveTo(250, 350)
+        this.context.lineTo(250, 25)
+        this.context.stroke()
+        break;
+      case 7:
+        this.context.beginPath()
+        this.context.moveTo(250, 25)
+        this.context.lineTo(400, 25)
+        this.context.stroke()
+        break;
+      case 6:
+        this.context.beginPath()
+        this.context.moveTo(400, 25)
+        this.context.lineTo(400, 70)         
+        this.context.stroke()
+        break;
+      case 5:
+        this.context.beginPath()
+        this.context.arc(400, 100, 30, 0, Math.PI * 2);
+        this.context.stroke()
+        break;
+      case 4:
+        this.context.beginPath()
+        this.context.moveTo(400, 130)
+        this.context.lineTo(400, 250)         
+        this.context.stroke()
+        break;
+      case 3:
+        this.context.beginPath()
+        this.context.moveTo(400, 250)
+        this.context.lineTo(360, 320)         
+        this.context.stroke()
+        break;
+      case 2:
+        this.context.beginPath()
+        this.context.moveTo(400, 250)
+        this.context.lineTo(440, 320)         
+        this.context.stroke()
+        break;
+      case 1:
+        this.context.beginPath()
+        this.context.moveTo(400, 170)
+        this.context.lineTo(320, 100)         
+        this.context.stroke()
+        break;
+      case 0:
+        this.context.beginPath()
+        this.context.moveTo(400, 170)
+        this.context.lineTo(480, 100)         
+        this.context.stroke()
+      
+        break;
+    }
   }
 
+  
+
   gameOver() {
-    // ... your code goes here
+      // console.log('yes, I am being called')
+      let gameOverImg = new Image()  
+      gameOverImg.src = './images/gameover.png' 
+      gameOverImg.onload = function (){
+        // console.log('test')
+        this.context.drawImage(gameOverImg, 0, 0)
```

Comment:
this.context is undefined here. Try to add in the previous line `      this.context = document.getElementById('hangman').getContext('2d')
` and it should work